pkgs/development/interpreters/octave/default.nix

```diff
@@ -93,6 +97,16 @@ let
   else
     null
   ;
+
+  octavePackages = import ../../../top-level/octave-packages.nix {
+    inherit pkgs;
+    inherit (pkgs) lib stdenv fetchurl newScope;
+    octave = self;
+    inherit blas lapack gfortran python texinfo gnuplot;
```

Comment:
You do use the same deps as octave here, but not accurately. Notice the difference between for example `blas` and `blas'`. I think it'd be best to not pass so many values here, and pass only `octave = self`, and put inherit the deps in the `callPackage` definition in `octave-packages.nix`, as explained in previous comment.

Reply:
See https://github.com/NixOS/nixpkgs/pull/108562/commits/b58f29416eee44575e039d463c7cb665555416e9